index.py

```diff
@@ -304,28 +341,29 @@ def generate_string_of_posting(self):
 
 class PostingList:
     """
-    Each PostingList is for a term, reflecting its term frequency in a number of document(s)
-    A PostingList contains size and Postings, which have doc_id and freq of each term
+    Each PostingList is a collection of Postings for a particular term.
+    A PostingList contains the number of unique documents it contains regardless of which zone/field (size) and a list of Postings (postings)
     """
     def __init__(self):
         self.postings = []
-        self.size = 0  # number of insert / term frequency
+        self.unique_docids = 0
 
     def get_size(self):
-        return self.size
+        return self.unique_docids
 
     # Insert with var byte encoding
-    def insert(self, doc_id, field, positions):
-        next_id = self.size
+    def insert(self, doc_id, field, positions, new_doc_id=True):
+        next_id = self.unique_docids
         new_posting = Posting(next_id, doc_id, field, positions)
         new_posting.var_byte_encoding()
         self.postings.append(new_posting)
-        self.size += 1
+        if new_doc_id:
+            self.unique_docids += 1
 
     def insert_without_encoding(self, doc_id, field, positions):
-        next_id = self.size
+        next_id = self.unique_docids
         self.postings.append(Posting(next_id, doc_id, field, positions))
-        self.size += 1
+        self.unique_docids += 1
```

Comment:
We should do a check here, whether or not to increment `unique_docids`, like the one done in `insert` function